test/functional/tests/conftest.py

```diff
@@ -156,25 +156,55 @@ def base_prepare(item):
         kill_all_io()
         DeviceMapper.remove_all()
 
-        if installer.check_if_installed():
-            try:
-                from api.cas.init_config import InitConfig
-                InitConfig.create_default_init_config()
-                unmount_cas_devices()
-                casadm.stop_all_caches()
-                casadm.remove_all_detached_cores()
-            except Exception:
-                pass  # TODO: Reboot DUT if test is executed remotely
-
-        for disk in TestRun.dut.disks:
-            disk.umount_all_partitions()
-            if not create_partition_table(disk, PartitionTable.gpt):
-                raise Exception(f"Failed to remove partitions from {disk}")
-
-        if get_force_param(item) and not TestRun.usr.already_updated:
-            installer.reinstall_opencas()
-        elif not installer.check_if_installed():
-            installer.install_opencas()
+        if list(TestRun.item.iter_markers(name="without_install")):
+            if installer.check_if_installed():
+                try:
+                    from api.cas.init_config import InitConfig
+                    InitConfig.create_default_init_config()
+                    unmount_cas_devices()
+                    casadm.stop_all_caches()
+                    casadm.remove_all_detached_cores()
+                except Exception:
+                    pass  # TODO: Reboot DUT if test is executed remotely
+                installer.uninstall_opencas()
+
+            for disk in TestRun.dut.disks:
+                disk.umount_all_partitions()
+                if not create_partition_table(disk, PartitionTable.gpt):
+                    raise Exception(f"Failed to remove partitions from {disk}")
```

Comment:
Apparently half of the code in both `if` and `else` blocks is exactly the same. Why not just move it before `if-else` block?